ic-os/dev-tools/bare_metal_deployment/benchmark_driver.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+set -eEuo pipefail
+
+benchmark_runner="${1}"
+ssh_key="${2}"
+ip_address="${3}"
+tools="${@:4}"
+
+key_component=()
+if [ "${ssh_key}" != "None" ]; then
+    key_component=("-i", "${ssh_key}")
+fi
+
+TMPDIR=$(mktemp -d)
+trap "rm -rf ${TMPDIR}" exit
+
+# Send over the runner
+scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null "${key_component[@]}" "${benchmark_runner}" "admin@[${ip_address}]:benchmark_runner.sh"
+
+# Send over the tools
+if [ "${tools}" != "" ]; then
+    scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null "${key_component[@]}" ${tools[@]} "admin@[${ip_address}]:"
+fi
```

Comment:
Shouldn't the test fail if benchmark_runner or tools are not found?

Reply:
I was trying to keep things really light for this first iteration. Right now, the entire benchmarking step can fail without blocking the pipeline. I think, we should move these components away from scripts when we have a chance, rather than building them out further.